mdf.md

```diff
@@ -0,0 +1,82 @@
+
+##  Table of Contents
+
+1. [Abstract](#abstract)
+2. [Format](#format)
+    1. [Fields](#fields)
+1. [Usage](#usage)
+    1. [Informational Fields](#informational-fields)
+    2. [Configurational Fields](#configurational-fields)
+
+## Abstract
+
+In this specification, we describe a method to construct both a linear and DAG (Directed Acyclic Graph) based message history that will aid the consistency of [MVDS](./mvds.md). Additionally we explain how data sync can be used for more lightweight messages that do not require full synchronization. This specification extends the [MVDS message](./mvds.md#payloads) to modify the functionality of MVDS.
+
+## Format
+
+We introduce the metadata message which is used to convey information about a message and how it SHOULD be handled. If an MVDS node implements this, we say that node has MDF capability (This might become part of a future reconciled MVDS spec).
+
+
+```protobuf
+package vac.mvds;
+
+message Metadata {
+  bytes parent = 7000;
+  repeated bytes previous_messages = 7001;
+  bool ack_required = 7002 [default = true];
+}
+```
+
+We MAY transmit a `Metadata` message by extending the MVDS [message](./mvds.md#payloads) with a `metadata` field.
+
+```diff
+message Message {
++ Metadata metadata = 6000;
+  bytes group_id = 6001;
+  int64 timestamp = 6002;
+  bytes body = 6003;
+}
+```
+### Fields
+
+| Name                   |   Description                                                                                                                    |
+| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| `parent`               |   contains the [`message indentifier`](./mvds.md#payloads) of the last sent message for the given node in the specific group id. |            
+| `previous_messages`    |   contains a list of parent [`message identifiers`](./mvds.md#payloads).                                                         |
+| `ack_required`         |   contains a flag whether a message needs to be acknowledged or not.                                                             |
+
+## Usage
+
+The flags provided through the `Metadata` message are either informational or behavioral. Informational fields MAY be used by a node, configurational fields SHOULD be recognized and the nodes behavior SHOULD change accordingly.
+
+### Informational Fields
+
+Below are the list of informational flags and what they MAY be used for by a node.
+
+#### `parent`
+
+This field contains the last sent message by `MVDS` for a specific group, it MUST NOT use any messages as parent whose `ack` flag was set to `false`. This creates a linked list of persistent messages.
+
+#### `previous_messages`
+
+This field contains a list of a messages previously sent or received by a node in a specific group. This helps establish ordering by creating a Directed Acyclic Graph (DAG)<sup>1</sup>.
```

Comment:
Is this an ordered list (asc/desc)? What kind of ordering are we achieving? If total, how are ties broken? Are these ancestors? Is this an unbound list (all ancestors from the beginning on the conversation) or it can be bound?

Reply:
I'd be more explicit re casual consistency, with some hints of how an (~arbitrary? forgot what the right term is) total order can be achieved. Similar to how we break ties in existing protocol

More explicit re which messages these are, I assume it is just previous layer? So can have similar limit to what git or matrix have, or just leave it as as a constant to be defined later.

Reply:
Added that applications should determine how many messages are to be included, additionally mentioned that messages MUST be ordered from latest to oldest.

Reply:
There are still some unanswered questions:

Are these just parents (1 level) , or ancestors (1+ level)? If that's left for the application to specify best to be explicit.
Ordered by what? What kind of ordering do we achieve? 
How do we use this field? 

> the messages MUST be ordered from latest to oldest sent or received.

This seems to imply that we use sent/received time, which is likely not what we want, given that there's no guarantee that messages will be sent/receive in order.

Reply:
> Are these just parents (1 level) , or ancestors (1+ level)? If that's left for the application to specify best to be explicit.

I'll specify but I assume applications will use ancestors so any x messages before the message in question. 

> This seems to imply that we use sent/received time, which is likely not what we want, given that there's no guarantee that messages will be sent/receive in order.

Should ordering be left to the application too?

Reply:
@cammellos i don’t know if the application will need to keep track, the app can ask data sync which already keeps track of the ids. But I see your point.